src/Umbraco.Core/Persistence/PetaPoco.cs

```diff
@@ -224,13 +226,18 @@ private void CommonConstruct()
 		// Automatically close one open shared connection
 		public void Dispose()
 		{
-			// Automatically close one open connection reference
-			//  (Works with KeepConnectionAlive and manually opening a shared connection)
-			CloseSharedConnection();
+		    Dispose(true);

```

Comment:
if we're using the pattern of Dispose(bool disposing) shouldn't this then implement the finalizer pattern and use 'false'?